tribe-ext-pdf-tickets/index.php

```diff
@@ -0,0 +1,633 @@
+<?php
+/**
+ * Plugin Name:     Event Tickets Extension: PDF Tickets
+ * Description:     RSVP, WooCommerce, and Easy Digital Downloads will become PDFs (will be saved to your Uploads directory) and attached to the ticket email.
+ * Version:         1.0.0
+ * Extension Class: Tribe__Extension__PDF_Tickets
+ * Author:          Modern Tribe, Inc.
+ * Author URI:      http://m.tri.be/1971
+ * License:         GPLv2 or later
+ * License URI:     https://www.gnu.org/licenses/gpl-2.0.html
+ */
+
+// Do not load unless Tribe Common is fully loaded.
+if ( ! class_exists( 'Tribe__Extension' ) ) {
+	return;
+}
+
+/**
+ * Extension main class, class begins loading on init() function.
+ */
+class Tribe__Extension__PDF_Tickets extends Tribe__Extension {
+
+	/**
+	 * The custom field in which to store the attendee's PDF ticket's file name.
+	 *
+	 * We generate a random file name for security purposes, but we store it in
+	 * the database for lookup purposes. It should not end in ".pdf" because
+	 * the file extension gets added manually during output of the full file
+	 * path or the full URL/link.
+	 *
+	 * @var string
+	 */
+	public $pdf_ticket_meta_key = '_tribe_tickets_pdf_file_name';
+
+	/**
+	 * The query argument key for the Attendee ID.
+	 *
+	 * @var string
+	 */
+	public $pdf_url_query_arg_key = 'tribe_tickets_pdf_attendee';
+
+	/**
+	 * The query argument key for retrying loading an attempted PDF.
+	 *
+	 * @var string
+	 */
+	public $pdf_retry_url_query_arg_key = 'tribe_tickets_pdf_retried';
+
+	/**
+	 * An array of the absolute file paths of the PDF(s) to be attached
+	 * to the ticket email.
+	 *
+	 * One PDF attachment per attendee, even in a single order.
+	 *
+	 * @var array
+	 */
+	protected $attachments_array = array();
+
+	/**
+	 * Setup the Extension's properties.
+	 *
+	 * This always executes even if the required plugins are not present.
+	 */
+	public function construct() {
+		$this->add_required_plugin( 'Tribe__Tickets__Main', '4.5.2' );
+
+		add_action( 'tribe_plugins_loaded', array( $this, 'required_tribe_classes' ), 0 );
+
+		$this->set_url( 'https://theeventscalendar.com/extensions/pdf-tickets/' );
+	}
+
+	/**
+	 * Check required plugins after all Tribe plugins have loaded.
+	 */
+	public function required_tribe_classes() {
+		$actives = Tribe__Dependency::instance()->get_active_plugins();
+
+		if ( array_key_exists( 'Tribe__Tickets_Plus__Main', $actives ) ) {
```

Comment:
Just for future reference checkout `\Tribe__Dependency::is_plugin_version()`. It helps with stuff like this.

Reply:
done now, since it was quick and easy.